tools/codegen/codespec/config.go

```diff
@@ -0,0 +1,93 @@
+package codespec
+
+import (
+	"strings"
+
+	"github.com/mongodb/terraform-provider-mongodbatlas/tools/codegen/config"
+)
+
+func applyConfigSchemaOptions(resourceConfig *config.Resource, resource *Resource) {
+	applySchemaOptions(resourceConfig.SchemaOptions, &resource.Schema.Attributes, "")
+}
+
+func applySchemaOptions(schemaOptions config.SchemaOptions, attributes *Attributes, parentName string) {
+	ignoredAttrs := getIgnoredAttributesMap(schemaOptions.Ignores)
+
+	var finalAttributes Attributes
+
+	for i := range *attributes {
+		attr := &(*attributes)[i]
+		attrPathName := getAttributePathName(string(attr.Name), parentName)
+
+		if shouldIgnoreAttribute(attrPathName, ignoredAttrs) {
+			continue
+		}
+
+		// the config is expected to use alias name for defining any subsequent overrides (description, etc)
+		applyAlias(attr, &attrPathName, schemaOptions)
```

Comment:
just to confirm, are we supporting alias in both path and in body attributes?  that's great as HC tool only supports aliases in path attributes

Reply:
by body, you mean inner nested attributes right? Then yes we are supporting that. I've added some test cases to cover that.